tests/models/encoder_decoders/nlvr/nlvr_coverage_semantic_parser_test.py

```diff
@@ -52,3 +58,46 @@ def test_forward_with_epoch_num_changes_cost_weight(self):
         # The config file has ``wait_num_epochs`` set to 0, so the model starts decreasing the cost
         # weight at epoch 0 itself.
         assert_almost_equal(cost_weights, [0.72, 0.648, 0.5832, 0.52488])
+
+    def test_initialize_weights_from_archive(self):
+        original_model_parameters = self.model.named_parameters()
+        original_model_weights = {name: parameter.data.clone().numpy()
+                                  for name, parameter in original_model_parameters}
+        # pylint: disable=line-too-long
+        mml_model_archive_file = "tests/fixtures/encoder_decoder/nlvr_direct_semantic_parser/serialization/model.tar.gz"
+        archive = load_archive(mml_model_archive_file)
+        archived_model_parameters = archive.model.named_parameters()
+        self.model._initialize_weights_from_archive(archive)
+        changed_model_parameters = dict(self.model.named_parameters())
+        for name, archived_parameter in archived_model_parameters:
+            archived_weight = archived_parameter.data.numpy()
+            original_weight = original_model_weights[name]
+            changed_weight = changed_model_parameters[name].data.numpy()
+            with self.assertRaises(AssertionError):
```

Comment:
I don't understand what these two lines are trying to do.  You want the original weights and the new weights to be different?  I guess this works, though it's an odd way to go about it.  A simple comment above this line would be helpful.

Reply:
I'm trying to check if the weights have indeed changed after calling `_initialize_weights_from_archive`. Added a comment.